Automattic/jetpack
# Add/protect status class

#### Changes proposed in this Pull Request:

* This PR creates the Status class in the Protect plugin. This class is responsible for fetching the current status of vulnerabilities check from WPCOM and cache it. It also adds the status to the initial state of the JS app.

#### Jetpack product discussion


p1HpG7-fgl-p2#comment-53520

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* To test this PR, you'll need to apply a patch to your sandbox and then sandbox your site
* Apply D78234-code to your sandbox
* Sandbox your site
* Read the ["Developing" section of the README added in this PR](https://github.com/Automattic/jetpack/blob/548c4c69a19f8935f04aa40a06d20517df6b9f1f/projects/plugins/protect/README.md#developing)
* Define the constants
* Visit Protect page and connect
* inspect `jetpackProtectInitialState` global variable in your browser and check the value of `status`
* Play with the constants described in the README and see if you get the expected results
